lib/MetaCPAN/Web/Role/Elasticsearch/Adapter.pm

```diff
@@ -0,0 +1,50 @@
+package MetaCPAN::Web::Role::Elasticsearch::Adapter;
+
+use Moose::Role;
+
+=head1 METHODS
+
+=head2 extract_first_element
+
+Elasticsearch 1.x changed the data structure returned when fields are used.
+For example before one could get a ArrayRef[HashRef[Str]] where now
+that will come in the form of ArrayRef[HashRef[ArrayRef[Str]]]
+
+This function will select the first element given an array of hash of array.
+So this:
+
+    [
+      {
+        distribution => ['WhizzBang'],
+        author       => ['EVE'],
+      },
+       ...
+   ]
+
+becomes:
+
+    [
+      {
+        distribution => 'WhizzBang',
+        author       => 'EVE',
+      },
+      ...
+    ]
+
+=cut
+
+sub extract_first_element {
+    my ( $self, $array ) = @_;
+    foreach my $hash ( @{$array} ) {
+        foreach my $field ( %{$hash} ) {
+            my $value = $hash->{$field};
+
+            # We only extract the first element when have an ArrayRef
+            next if not( ref($value) and ( ref($value) eq 'ARRAY' ) );
```

Comment:
I don't think a runtime error is the answer...
This is intended for specific queries (and explicitly not useful for specific other queries),
so the issue as I see it is the likelihood for the developer to make a mistake.

If the developer is going to accidentally use this when they shouldn't, then it seems equally likely to me that they might test it on a value that only has one element, but if there's a value in production that has multiple, we'll be serving 500's.

Which is better?  A mysterious bug that shows only one of multiple values, or serving errors to the user?

Perhaps we should consider making the code forward compatible instead of backward compatible?